src/block.rs

```diff
@@ -552,61 +552,93 @@ impl Block {
 }
 
 #[derive(Copy, Clone, Debug, PartialEq)]
-pub enum Comparator {
+pub enum Operator {
+    /// Can perhaps be replaced by wrapping the operator in an option.
     None,
-    /// Eq is also Assign
+    /// The = operator. Can be used as assignment, or to open a scope.
     Eq,
-    /// The == operator, which means Eq but cannot be used to assign
-    EEq,
-    /// The ?= operator
-    QEq,
-    Lt,
-    Gt,
-    Le,
-    Ge,
-    Ne,
+    /// The ?= operator Can be used to open a scope on the condition that it exists.
```

Comment:
?= can also be used as a conditional comparator. `scope:faith ?= faith:catholicism` is an equality comparison with a short-circuit to false if scope:faith does not exist. The main difference with = is that it won't log an error. Unfortunately, modders have been enthusiastically using ?= everywhere to hide their errors.

Reply:
I didn't know ?= could also be used as a conditional equality comparator. That sheds a new light on things, Comparator now makes more sense as a name for the enum. I have changed it.